src/main/java/com/google/research/bleth/simulator/StatisticsState.java

```diff
@@ -0,0 +1,109 @@
+     * has been observed by at least one observer, i.e. has been detected by the resolver.
+     */
+    public void writeBeaconsObservedPercentStats() {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        for (String beaconId : beaconsObservedPercent.keySet()) {
+            Entity entity = new Entity(Schema.StatisticsState.entityKindBeaconsObservedPercent);
+            entity.setProperty(Schema.StatisticsState.simulationId, simulationId);
+            entity.setProperty(Schema.StatisticsState.beaconId, beaconId);
+            entity.setProperty(Schema.StatisticsState.percent, beaconsObservedPercent.get(beaconId));
+            datastore.put(entity);
+        }
+    }
+
+    /**
+     * Read from the db statical data about the distance between the beacons' real locations and their estimated locations.
+     * @param simulationId is the simulation id associated with the statistical data.
+     * @return a map that maps an aggregate function to its value on the simulation.
+     */
+    public static Map<String, Double> readDistancesStats(String simulationId) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        ImmutableMap.Builder<String, Double> distanceStats = new ImmutableMap.Builder<>();
+
+        Query.FilterPredicate filterBySimulationId =
+                new Query.FilterPredicate(Schema.StatisticsState.simulationId, Query.FilterOperator.EQUAL, simulationId);
+        PreparedQuery distances = datastore.prepare(new Query(Schema.StatisticsState.entityKindDistance).setFilter(filterBySimulationId));
+
+        for (Entity entity : distances.asIterable()) {
+            String kind = (String) entity.getProperty(Schema.StatisticsState.aggregateFunction);
+            Double value = (Double) entity.getProperty(Schema.StatisticsState.value);
+            distanceStats.put(kind, value);
+        }
+        return distanceStats.build();
+    }
+
+    /**
+     * Read from the db statical data about the percentage of rounds each beacon has been observed.
+     * @param simulationId is the simulation id associated with the statistical data.
+     * @return a map that maps a beacon's id to the percentage of rounds a beacon has been observed by at least one observer.
+     */
+    public static Map<String, Double> readBeaconsObservedPercentStats(String simulationId) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        ImmutableMap.Builder<String, Double> beaconsObserved = new ImmutableMap.Builder<>();
+
+        Query.FilterPredicate filterBySimulationId =
+                new Query.FilterPredicate(Schema.StatisticsState.simulationId, Query.FilterOperator.EQUAL, simulationId);
+        PreparedQuery percents = datastore.prepare(new Query(Schema.StatisticsState.entityKindBeaconsObservedPercent).setFilter(filterBySimulationId));
```

Comment:
Prefer to split this line (one for the query and one for the prepared query).
(Same applies at `readDistancesStats`).